### DISCUSSION TITLE
TheThingsNetwork MQTT Connector

### DISCUSSION THREAD

**Discussion by MacTelnet** [thingsboard/thingsboard#12320] [Q&A]
Read all the guides but I can't make TTN working.
With PE it works likes a charm, but with CE It doesn't add any device despite being connected and syncing.
Help please.

**Comment by K0I05** [comment]
Initially, it was trivial, but I managed to figure it out by reading the guides along with trial and error.

[MQTT Device API](https://thingsboard.io/docs/pe/reference/mqtt-api/)

Ensure your device is connecting to the correct topic, i.e. v1/devices/me/telemetry, for telemetry data.  And ensure that your message is formatted correctly: {"key1":"value1", "key2":"value2"} or [{"key1":"value1"}, {"key2":"value2"}] or {"ts":1451649600512, "values":{"key1":"value1", "key2":"value2"}}.  Note that the timestamp is epoch time in milliseconds.

In ThingsBoard, go to entities and then Devices, and then Manage Credentials.  Select MQTT Basic to start with and generate your keys (clientid, username, and password).  For MQTT basic, the URI should be: mqtt://thingsboard.cloud.

Likewise, you can test this using an MQTT client application, I used MQTTX to validate things before implementing changes on my device.

![devices](https://github.com/user-attachments/assets/78a996a7-31b2-484e-8f80-e20ecaf0e6b9)

**Reply by K0I05** [reply]
Sorry about that, I noticed the subject is ThingsNetwork, this is the ThingsBoard discussion board.  However, the general process should be applicable: it's been a few years since I've used ThingsNetwork.
